aws-connect-user/src/main/java/software/amazon/connect/user/UpdateHandler.java

```diff
@@ -68,9 +76,13 @@
                                                                                  final ProgressEvent<ResourceModel, CallbackContext> progress,
                                                                                  final CallbackContext context,
                                                                                  final Logger logger) {
-        final boolean updateUserEmail = !StringUtils.equals(desiredStateModel.getIdentityInfo().getEmail(), previousStateModel.getIdentityInfo().getEmail());
-        final boolean updateUserFirstName = !StringUtils.equals(desiredStateModel.getIdentityInfo().getFirstName(), previousStateModel.getIdentityInfo().getFirstName());
-        final boolean updateUserLastName = !StringUtils.equals(desiredStateModel.getIdentityInfo().getLastName(), previousStateModel.getIdentityInfo().getLastName());
+        final String desiredUpdateUserEmail = desiredStateModel.getIdentityInfo() == null ? null : desiredStateModel.getIdentityInfo().getEmail();
+        final String desiredUpdateUserFirstName = desiredStateModel.getIdentityInfo() == null ? null : desiredStateModel.getIdentityInfo().getFirstName();
+        final String desiredUpdateUserLastName = desiredStateModel.getIdentityInfo() == null ? null : desiredStateModel.getIdentityInfo().getLastName();
+
+        final boolean updateUserEmail = !StringUtils.equals(desiredUpdateUserEmail , previousStateModel.getIdentityInfo().getEmail());
```

Comment:
If the previous Model does not have Identity Info in the template, will that cause an NPE at update?